File: Java/InsertionSort.java
public class InsertionSort {

	static void result(int a[])
	{
		for(int i=1;i<a.length;i++)
		{
			int temp =a[i];
			int j=i-1;
			
			while(j>=0 && a[j]>temp)
			{
				a[j+1]=a[j];
				j=j-1;
			}
			a[j+1]=temp;
		}
	}
	
	
	public static void main(String[] args) {
		int a[]= {12,45,1,5,18,2,3};
		
		InsertionSort.result(a);
		for(int x:a) {
			System.out.print(x+" ");
		}

	}

}

File: Hacktoberfest-2023/DSA/insertionSort.cpp
// Problem Statement: Given an array of N integers, write a program to implement the Insertion sorting algorithm.

// Examples:


// Example 1:
// Input: N=5, array[] = {5,4,3,2,1}
// Output: 1,2,3,4,5
// Explanation: After sorting the array is: 1,2,3,4,5

  
// Example 2:
// Input: N = 6, array[] = {13,46,24,52,20,9}
// Output: 9,13,20,24,46,52
// Explanation: 
// After sorting the array is: 9,13,20,24,46,52


#include <bits/stdc++.h>
using namespace std;

void insertionSort(int arr[], int n) {
    for (int i = 0; i <= n - 1; i++) {
        int j = i;
        while (j > 0 && arr[j - 1] > arr[j]) {
            int temp = arr[j - 1];
            arr[j - 1] = arr[j];
            arr[j] = temp;
            j--;
        }
    }

    cout << "After sorting: " << "\n";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << "\n";
}

int main()
{
    int arr[] = {13, 46, 24, 52, 20, 9};
    int n = sizeof(arr) / sizeof(arr[0]);
    cout << "Before Sorting: " << endl;
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;

    insertionSort(arr, n);
    return 0;
}
